shared/src/commonMain/kotlin/com/vml/tutorial/plantshop/plants/data/PlantsRepository.kt

```diff
@@ -1,16 +1,55 @@
 package com.vml.tutorial.plantshop.plants.data
 
+import com.vml.tutorial.plantshop.core.utils.Logger
+import com.vml.tutorial.plantshop.plants.domain.DbDataSource
 import com.vml.tutorial.plantshop.plants.domain.Plant
 import com.vml.tutorial.plantshop.plants.domain.PlantsDataSource
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.launch
 
 interface PlantsRepository {
     fun getPlants(): List<Plant>
+    fun getFavorites(): Flow<List<Plant>>
+    suspend fun toggleFavoriteStatus(plantId: Int)
 }
 
 class PlantsRepositoryImpl(
-    private val localDataSource: PlantsDataSource
-): PlantsRepository {
+    private val localDataSource: PlantsDataSource,
+    private val dbDataSource: DbDataSource,
+    private val coroutineScope: CoroutineScope
```

Comment:
You don't need this here :) Usually, we don't pass coroutine scope in this layer or lower layers instead we just use suspend functions. 

Good for us -> because you do not need to pass an extra argument( the coroutine scope) at every lower level eg.: data, repository, use-case etc, which can leak because of the lifecycle state. 
You will call the proper lifecycleScope from viewModel or Component (from Decopmpose in our case) which will be lifecycle-aware.

hint:
Component has an extension function -> componentCoroutineScope() this will provide you a lifecycle-aware corotuine scope.

Reply:
Thank you for this information :) Unnecessary coroutine scopes are removed